linera-views/src/backends/lru_caching.rs

```diff
@@ -1,45 +1,305 @@
 // Copyright (c) Zefchain Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-//! Add LRU (least recently used) caching to a given store.
+//! Add caching to a given store.
```

Comment:
As for code organisation – I'd split this code into `mod.rs` + files for non-public elements. It would make reading (and understanding) much easier. For example, there's a lot of non-`pub` code here at the top of the file. So when reading it I read a lot of "implementation details" first, w/o even getting to what is the public facing API we're adding.